<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Call broad peak by Hidden Markov Models with t emissions — HMMtBroadPeak • HMMtBroadPeak</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Call broad peak by Hidden Markov Models with t emissions — HMMtBroadPeak" />
<meta property="og:description" content="Call very broad peaks for data such as LAD domains, NAD domains.
Reads will be count by each bins. Only bins with at least given reads
(defined by background parameter) for all samples (pool all reads for 
each bin) will be subsequently normalized. These bins will be first 
normalized to CPM (count per million) reads and then do log2 transform
for the ratio over control with a pseudocount. 
The peaks were defined by running a hidden markov model over the 
normalized values (using the R-package HMMt)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HMMtBroadPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/HMMtBroadPeak.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jianhong/HMMtBroadPeakissues/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Call broad peak by Hidden Markov Models with t emissions</h1>
    <small class="dont-index">Source: <a href='https://github.com/jianhong/HMMtBroadPeakissues/blob/master/R/HMMtBroadPeak.R'><code>R/HMMtBroadPeak.R</code></a></small>
    <div class="hidden name"><code>HMMtBroadPeak.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Call very broad peaks for data such as LAD domains, NAD domains.
Reads will be count by each bins. Only bins with at least given reads
(defined by background parameter) for all samples (pool all reads for 
each bin) will be subsequently normalized. These bins will be first 
normalized to CPM (count per million) reads and then do log2 transform
for the ratio over control with a pseudocount. 
The peaks were defined by running a hidden markov model over the 
normalized values (using the R-package HMMt).</p>
    </div>

    <pre class="usage"><span class='fu'>HMMtBroadPeak</span><span class='op'>(</span>
  <span class='va'>treatment</span>,
  <span class='va'>control</span>,
  binSize <span class='op'>=</span> <span class='fl'>5000</span>,
  background <span class='op'>=</span> <span class='fl'>10</span>,
  pseudocount <span class='op'>=</span> <span class='fl'>1</span>,
  gapwidth <span class='op'>=</span> <span class='va'>binSize</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>treatment, control</th>
      <td><p>Bam file of treatments and controls. Make sure the
index file keep same prefix name with bam file.</p></td>
    </tr>
    <tr>
      <th>binSize</th>
      <td><p>The size of bins for count</p></td>
    </tr>
    <tr>
      <th>background</th>
      <td><p>Only bins with at least background reads (pool all reads)
will be subsequently normalized.</p></td>
    </tr>
    <tr>
      <th>pseudocount</th>
      <td><p>default 1.</p></td>
    </tr>
    <tr>
      <th>gapwidth</th>
      <td><p>The Ranges of peaks separated by a gap less than gapwith 
positions will be merged.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>parameters passed to <a href='https://rdrr.io/pkg/HMMt/man/BaumWelchT.html'>BaumWelchT</a> 
except m (fixed to 2).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list with elements counts and peaks. Bothe counts and peaks 
are GRanges objects.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>treatment</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"LB1.KD.chr1_1_5000000.bam"</span>,
                          package <span class='op'>=</span> <span class='st'>"HMMtBroadPeak"</span>,
                          mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>control</span>   <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"LB1.WT.chr1_1_5000000.bam"</span>,
                          package <span class='op'>=</span> <span class='st'>"HMMtBroadPeak"</span>,
                          mustWork <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>called</span> <span class='op'>&lt;-</span> <span class='fu'>HMMtBroadPeak</span><span class='op'>(</span><span class='va'>treatment</span>, <span class='va'>control</span><span class='op'>)</span>
</div><div class='output co'>#&gt; iteration: 1iteration: 2iteration: 3iteration: 4iteration: 5iteration: 6iteration: 7iteration: 8iteration: 9iteration: 10iteration: 11iteration: 12iteration: 13iteration: 14iteration: 15iteration: 16iteration: 17iteration: 18iteration: 19iteration: 20iteration: 21iteration: 22iteration: 23iteration: 24iteration: 25iteration: 26iteration: 27iteration: 28iteration: 29iteration: 30iteration: 31iteration: 32iteration: 33iteration: 34iteration: 35iteration: 36iteration: 37iteration: 38iteration: 39iteration: 40iteration: 41iteration: 42iteration: 43iteration: 44iteration: 45iteration: 46iteration: 47iteration: 48iteration: 49iteration: 50iteration: 51iteration: 52iteration: 53</div><div class='input'><span class='va'>called</span><span class='op'>$</span><span class='va'>peaks</span>
</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;       seqnames          ranges strand
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr1  774227-1698303      *
#&gt;   [2]     chr1 1713289-2657344      *
#&gt;   [3]     chr1 2777225-5000001      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome</div><div class='input'><span class='fu'><a href='plotPeaks.html'>plotPeaks</a></span><span class='op'>(</span><span class='va'>called</span>, seqname<span class='op'>=</span><span class='st'>"chr1"</span><span class='op'>)</span>
</div><div class='img'><img src='HMMtBroadPeak-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


